src/integration/test-helpers.ts

```diff
@@ -123,3 +123,29 @@ export const waitUntilRecordsReady = async (
 
   return indexStats;
 };
+
+type Assertion = (result: any) => void;
+
+export const assertWithRetries = async (
```

Comment:
This is really great, thanks for adding this function. I have sent you some optional notes internally for this function about the numbers.